index.js

```diff
@@ -16,6 +16,7 @@ var through = require('through2')
 var util = require('util')
 
 var BASE_URI = 'https://www.googleapis.com/upload/storage/v1/b'
+var EXPIRED_UPLOAD_STATUS_CODE = 410
```

Comment:
s/EXPIRED/TERMINATED/

It could also be that the upload is unrecoverable for other reasons, e.g., a service error encountered on the attempt to finalize the object.

<peers at code> And actually I think we actually return 404 if the session has expired, so this is literally only for cases where the session died due to bad data or a service error.